sympy/functions/elementary/trigonometric.py

```diff
@@ -1745,6 +1761,9 @@ def fdiff(self, argindex=1):
         else:
             raise ArgumentIndexError(self, argindex)
 
+    def _eval_rewrite_as_besselj(self, arg, **kwargs):
+        from sympy.functions.special.bessel import besselj
+        return Piecewise(((sqrt(2)/(sqrt(pi)*sqrt(arg)*besselj(-S.Half, arg))), Ne(arg, 0)), (1, True))
```

Comment:
Case like this where we have `sqrt(arg)` need to be checked very carefully for branch cuts because `sqrt(x)` has a branch cut on the negative real axis and so does `besselj`. I think that all branch cuts here are correct but we need to be very careful about any possible differences like `1/sqrt(arg)` vs `sqrt(1/arg)` etc.

Here is how you can check the branch cuts:
```python
In [27]: tan(x).rewrite(besselj)
Out[27]: 
besselj(1/2, x) 
────────────────
besselj(-1/2, x)

In [28]: tan(x).rewrite(besselj).subs(x, -1).n()
Out[28]: -1.55740772465490

In [29]: tan(x).rewrite(besselj).subs(x, -1+1e-6j).n()
Out[29]: -1.55740772464957 + 3.42551882080531e-6⋅ⅈ

In [30]: tan(x).rewrite(besselj).subs(x, -1-1e-6j).n()
Out[30]: -1.55740772464957 - 3.42551882080531e-6⋅ⅈ

In [31]: tan(x).subs(x, -1).n()
Out[31]: -1.55740772465490

In [32]: tan(x).subs(x, -1+1e-6j).n()
Out[32]: -1.55740772464957 + 3.42551882080531e-6⋅ⅈ

In [33]: tan(x).subs(x, -1-1e-6j).n()
Out[33]: -1.55740772464957 - 3.42551882080531e-6⋅ⅈ
```